io/disk/ssd/nvmetest.py

```diff
@@ -54,6 +56,66 @@ def setUp(self):
         self.format_size = pow(2, int(self.format_size))
         cmd = "nvme id-ns %s | grep 'in use' | awk '{print $2}'" % self.id_ns
         self.lba = process.system_output(cmd, shell=True).strip('\n')
+        self.firmware_url = self.params.get('firmware_url', default='')
+        if 'firmware_upgrade' in str(self.name) and not self.firmware_url:
+            self.skip("firmware url not gien")
```

Comment:
I think we need not worry about that. The scope of this test is:
* Ensure firmware update works properly
* Ensure that by the end of the test, the controller is in the firmware given by the user.

I can take your suggestion as an enhancement, and I will work on it in the next commit.